src/main/java/ru/vk/itmo/pelogeikomakar/PermanentCompactableDao.java

```diff
@@ -0,0 +1,301 @@
+                loadTmpTable(tempTable);
+            } else {
+                makeCompactDone(tempTable, tempIndexPath, statFilePath);
+            }
+        }
+
+        Files.deleteIfExists(statFilePath);
+        Files.deleteIfExists(tempTablePath);
+        Files.deleteIfExists(tempIndexPath);
+
+    }
+
+    private void makeCompactDone(MemorySegment tempTable, Path tempIndexPath, Path statFilePath) throws IOException {
+        List<Long> dataOffsets = new ArrayList<>();
+        long offset = 0;
+        while (offset < tempTable.byteSize()) {
+            dataOffsets.add(offset);
+            long sizeOfKey = tempTable.get(ValueLayout.JAVA_LONG_UNALIGNED, offset);
+            long valueOffset = offset + Long.BYTES + sizeOfKey;
+            long valueSize = tempTable.get(ValueLayout.JAVA_LONG_UNALIGNED, valueOffset);
+
+            offset = valueOffset + Long.BYTES + valueSize;
+        }
+        long indexTableSize = (long) dataOffsets.size() * Long.BYTES;
+
+        try (var arenaWriter = Arena.ofShared(); var fileIndexOut = FileChannel.open(tempIndexPath,
+                StandardOpenOption.READ, StandardOpenOption.WRITE, StandardOpenOption.CREATE)) {
+
+            MemorySegment memSegmentIndexOut = fileIndexOut.map(FileChannel.MapMode.READ_WRITE,
+                    0, indexTableSize, arenaWriter);
+
+            long offsetIndex = 0;
+            for (var dataPtr : dataOffsets) {
+                memSegmentIndexOut.set(ValueLayout.JAVA_LONG_UNALIGNED, offsetIndex, dataPtr);
+                offsetIndex += Long.BYTES;
+            }
+
+        }
+
+        int oldTablesNumber;
+        try (var dis = new DataInputStream(new FileInputStream(statFilePath.toString()))) {
+            oldTablesNumber = dis.readInt();
+        }
+
+        deleteOldFiles(false, oldTablesNumber);
+
+        Path sourceTable = daoConfig.basePath()
+                .resolve(SSTABLE_NAME + ZERO_TMP);
+        Path targetTable = daoConfig.basePath()
+                .resolve(SSTABLE_NAME + Integer.toString(0));
+
+        Path sourceIndex = daoConfig.basePath()
+                .resolve(INDEX_NAME + ZERO_TMP);
+        Path targetIndex = daoConfig.basePath()
+                .resolve(INDEX_NAME + Integer.toString(0));
+
+        renameFile(sourceTable, targetTable);
+        renameFile(sourceIndex, targetIndex);
+    }
+
+    private void loadTmpTable(MemorySegment tempTable) {
+        long offset = 0;
+        while (offset < tempTable.byteSize()) {
+            long sizeOfKey = tempTable.get(ValueLayout.JAVA_LONG_UNALIGNED, offset);
+            MemorySegment currKey = tempTable.asSlice(offset + Long.BYTES, sizeOfKey);
+
+            long valueOffset = offset + Long.BYTES + sizeOfKey;
+            long valueSize = tempTable.get(ValueLayout.JAVA_LONG_UNALIGNED, valueOffset);
+
+            MemorySegment currValue = tempTable.asSlice(valueOffset + Long.BYTES, valueSize);
+            var entry = new BaseEntry<MemorySegment>(currKey, currValue);
```

Comment:
Зачем тратить символы и IDEA подсвечивает:

```suggestion
            var entry = new BaseEntry<>(currKey, currValue);
```